### ISSUE TITLE
Error when amending existing resource due to use of 'kubectl create'

### ISSUE THREAD

**Issue by dgard1981** [aws-quickstart/quickstart-kubernetes-resource-provider#15]
The error below specifies that `storageclasses.storage.k8s.io "gp2" already exists` already exists. This is absolutely true, but it's a default resource that was provisioned when the EKS cluster was created, and it's a resource that I wish to amend.

However, looking at the logs, it looks like `kubectl create` is being used as opposed to `kubectl apply`, meaning that it's not possible to amend existing resources using the `AWSQS::Kubernetes::Resource` resource type.

Is it possible to update the resource type so that either of the following is true?
1. Users specify an optional parameter in CloudFormation, e.g. `Mode`, with either `create` (default) or `apply`.
2. `kubectl apply` is used by default.

### Error from CloudWatch

```
executing command: kubectl create --save-config -o yaml -f /tmp/manifest.yaml -n default
Command failed with exit code 1, stderr: Error from server (AlreadyExists): error when creating "/tmp/manifest.yaml": storageclasses.storage.k8s.io "gp2" already exists
```

**Comment by prashilgupta** [comment]
+1
We are seeing same issue in another resource for Calico. Did you find solution for it?

**Comment by dgard1981** [comment]
@prashilgupta no, I didn't find a solution. It doesn't look like anyone is actively working on AWSQS projects as tickets are not even acknowledged, so I abandoned use of it.

**Comment by tlindsay42** [comment]
For posterity, EKS does not plan to change this built-in feature for the foreseeable future and we have no plans to build a workaround.

https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi-migration-faq.html

> #### What is the default StorageClass applied in newly created Amazon EKS version 1.23 and later clusters?
>
> The default StorageClass behavior remains unchanged. With each new cluster, Amazon EKS applies a kubernetes.io/aws-ebs based StorageClass named gp2. We don't plan to ever remove this StorageClass from newly created clusters. Though subject to change, we are currently targeting version 1.25 to remove the default annotation from the StorageClass applied to newly created clusters. Separate from the cluster default StorageClass, if you create an ebs.csi.aws.com based StorageClass without specifying a volume type, the Amazon EBS CSI driver will default to using gp3.

**Comment by dgard1981** [comment]
@tlindsay42, I think you've missed the point here. I wasn't asking  not asking for EKS's behaviour to change, but rather for the behaviour of the `AWSQS::Kubernetes::Resource` type to change. The example I used merely mentioned a resource provisioned by EKS.

However, because the `AWSQS::Kubernetes::Resource` type uses `create` instead of apply, it means that a CF changeset cannot be used to update any existing resource, even one previously created by the underlying template.

Basically, unless you plan to create a K8s resource using this type and then never ever touch it ever again, the type is unusable and that is why I have abandoned it's use.

**Comment by tlindsay42** [comment]
@dgard1981 Got it. Followed up with jaymccon@ and learned that the design decision behind this is:

> kubectl apply will update existing resources which breaks the cfn registry contract. kubectl create fails when the resources in your manifest already exist, which is what cfn expects. (https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html#resource-type-test-contract-additional-leaking)

Given this constraint of the CloudFormation Registry, we can't use apply in CloudFormation create resource operations: https://github.com/aws-quickstart/quickstart-kubernetes-resource-provider/blob/902c1a6ee2531cad8292cb5bfbde17acdeb4716b/apply/src/awsqs_kubernetes_resource/handlers.py#L68, but we do use kubectl apply in update resource operations: https://github.com/aws-quickstart/quickstart-kubernetes-resource-provider/blob/902c1a6ee2531cad8292cb5bfbde17acdeb4716b/apply/src/awsqs_kubernetes_resource/handlers.py#L128. I'll add a note for future reference.
